### DISCUSSION TITLE
How do I check whether a package toolset is clang or apple-clang?

### DISCUSSION THREAD

**Discussion by xq114** [xmake-io/xmake#2190] [Q&A]
目前在package中写package:has_tool("cxx", "clang")会把apple-clang和clang混为一谈；能否将两者区分开？即package:has_tool("cxx", "clang")仅对clang生效，package:has_tool("cxx", "clang", "appleclang")这样才把apple clang也包括在内。appleclang现在和clang已经有很多地方不一样了，一些新包必须要针对不同编译器设置

**Comment by waruqi** [comment]
has_tool 没法简单分开，改动太大，而且即使分开了，内部很多现有的 has_tool clang 逻辑都会被 broken 

你只能自己运行 os.iorunv clang --version 去判断下

**Reply by xq114** [reply]
如果要考虑兼容问题，可以这么改：has_tool("cxx", "clang")包括clang和apple clang，而仅包括clang可以用`has_tool("cxx", "clang") and not has_tool("cxx", "appleclang")`这样原来的包也不会有问题

**Reply by waruqi** [reply]
has_tool 目前对应的 toolname ，以 core/tools/clang.lua 来的，要么全分 加个 appleclang.lua ，不存在 or 关系，如果又要兼容，又要分开，还是得大改，目前也不太好改。
